Create 'Applications' section of the Controller UG (#510)
ansible/aap-docs: downstream/assemblies/platform/assembly-controller-applications.adoc

[id="assembly-controller-applications"]

= Applications

Creating and configuring token-based authentication for external applications such as ServiceNow and Jenkins makes it easier for them to integrate with {Controllername}.

OAuth 2 enables you to use tokens to share data with an application without disclosing login information. 
These tokens can also be scoped as "read-only". 

You can create an application that is representative of the external application you are integrating with, then use it to create tokens for the application to use on behalf of its users.

Associating these tokens with an application resource enables you to manage all tokens issued for a particular application. 
By separating the issue of tokens under *Applications*, you can revoke all tokens based on the Application without having to revoke all tokens in the system.

When integrating an external web application with {ControllerName} the web application may need to create OAuth2 Tokens on behalf of users of the web application. 

> Creating an application with the Authorization Code grant type is the preferred way to do this because:

Reviewer: ```suggestion
Creating an application with the Authorization Code grant type is the preferred way to do this for the following reasons:
```
Complete sentence for lead-in text

> * external applications can obtain a token for users, using their credentials

Reviewer: ```suggestion
* External applications can obtain a token for users, using their credentials
```

> * compartmentalized tokens issued for a particular application, enables those tokens to be easily managed. 

Reviewer: ```suggestion
* Compartmentalized tokens issued for a particular application, enables those tokens to be easily managed. 
```
